crates/sui-types/src/temporary_store.rs

```diff
@@ -293,64 +266,6 @@ impl<S> TemporaryStore<S> {
         }
     }
 
-    /// Compute storage gas for each mutable input object (including the gas coin), and each created object.
-    /// Compute storage refunds for each deleted object
-    /// Will *not* charge any computation gas. Returns the total size in bytes of all deleted objects + all mutated objects,
-    /// which the caller can use to charge computation gas
-    fn charge_gas_for_storage_changes(
-        &mut self,
-        gas_status: &mut SuiGasStatus<'_>,
-        gas_object_id: ObjectID,
-    ) -> Result<u64, ExecutionError> {
-        let mut total_bytes_written_deleted = 0;
-
-        // If the gas coin was not yet written, charge gas for mutating the gas object in advance.
-        let gas_object = self
-            .read_object(&gas_object_id)
-            .expect("We constructed the object map so it should always have the gas object id")
-            .clone();
-        self.written
-            .entry(gas_object_id)
-            .or_insert_with(|| (gas_object, WriteKind::Mutate));
-        self.ensure_active_inputs_mutated();
-        let mut objects_to_update = vec![];
-
-        for (object_id, (object, write_kind)) in &mut self.written {
-            let (old_object_size, storage_rebate) = self
-                .input_objects
-                .get(object_id)
-                .map(|old| (old.object_size_for_gas_metering(), old.storage_rebate))
-                .unwrap_or((0, 0));
-
-            let new_object_size = object.object_size_for_gas_metering();
-            let new_storage_rebate =
-                gas_status.charge_storage_mutation(new_object_size, storage_rebate.into())?;
-            object.storage_rebate = new_storage_rebate;
-            if !object.is_immutable() {
-                objects_to_update.push((object.clone(), *write_kind));
-            }
-            total_bytes_written_deleted += old_object_size + new_object_size;
-        }
-
-        for object_id in self.deleted.keys() {
-            // If an object is in `self.deleted`, and also in `self.objects`, we give storage rebate.
-            // Otherwise if an object is in `self.deleted` but not in `self.objects`, it means this
-            // object was unwrapped and then deleted. The rebate would have been provided already when
-            // mutating the object that wrapped this object.
-            if let Some(old_object) = self.input_objects.get(object_id) {
```

Comment:
This is bug 2: similarly, if `object_id` is a dynamic field with a storage rebate, we'll throw away that storage rebate here.